tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -52,6 +52,24 @@ static PyObject * THPVariable_contiguous(PyObject* self, PyObject* args)
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject * THPVariable_copy_(PyObject* self, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({ "copy_(Tensor other)" });
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
```

Comment:
Not sure what the semantics are here, but are we guaranteed that `self` is a THPVariable? i.e. why is there not a `THPVariable_Check(...)` here?